flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -324,39 +326,39 @@ public void setInputChannel(IntermediateResultPartitionID partitionId, InputChan
 		}
 	}
 
-	public void updateInputChannel(InputChannelDeploymentDescriptor icdd) throws IOException, InterruptedException {
+	public void updateInputChannel(
+			ResourceID localLocation,
+			PartitionInfo partitionInfo) throws IOException, InterruptedException {
 		synchronized (requestLock) {
 			if (closeFuture.isDone()) {
 				// There was a race with a task failure/cancel
 				return;
 			}
 
-			final IntermediateResultPartitionID partitionId = icdd.getConsumedPartitionId().getPartitionId();
+			ShuffleDescriptor shuffleDescriptor = partitionInfo.getShuffleDescriptor();
+			checkArgument(shuffleDescriptor instanceof NettyShuffleDescriptor,
```

Comment:
the `checkArgument` and the transformation could be put under the below `if` clause.

Reply:
True, I will put it under if condition, but `isLocalTo` is a detail of netty implementation and `PartitionInfo` belongs to general shuffle API.